### PR TITLE
(PPS-261): (feat) Add service-info endpoint to DRS 

### PR DESCRIPTION
**Pull request by BinamB**
https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.3.0/docs/#tag/Service-Info
### New Features
- Add GA4GH DRS `/service-info` endpoint

### CODE REVIEW

File: indexd/default_settings.py [hunk 1]
```diff
@@ -48,7 +49,17 @@
         "name": "DRS System",
         "host": "https://example.com/api/ga4gh/drs/v1/",
         "hints": [],
-        "type": "drs",
+        "type": {
+            "group": "org.ga4gh",
+            "artifact": "drs",
+            "version": "1.0.0",
+        },
+        "version": "1.0.0",
```
Comment (MichaelLukowski) [inline indexd/default_settings.py:57]: Can we bump this up to 1.0.3

File: indexd/drs/blueprint.py [hunk 2]
```diff
@@ -1,14 +1,65 @@
...
+                and dist["type"]["artifact"] == "drs"
+            ):
+                drs_dist = dist
+        if drs_dist == {}:
+            drs_dist = blueprint.dist[0]
+
+    ret = {
+        "id": drs_dist.get("id", reverse_domain_name),
+        "name": drs_dist.get("name", "DRS System"),
+        "version": drs_dist.get("version", "1.0.0"),
```
Comment (MichaelLukowski) [inline indexd/drs/blueprint.py:42]: Same can we bump the version here

File: indexd/utils.py [hunk 3]
```diff
@@ -189,3 +189,21 @@ def migrate_database(driver, migrate_functions, current_schema_version, model):
             f(engine=driver.engine, session=s)
             schema_version.version += 1
             s.add(schema_version)
+
+
+def drs_service_info_id_url_reversal(url):
+    """
+    Reverse the domain name for drs service-info IDs
```
Comment (paulineribeyre) [inline indexd/utils.py:196]: can we just call this function `reverse_url()` since it's pretty generic?

File: indexd/drs/blueprint.py [hunk 4]
```diff
@@ -1,14 +1,65 @@
...
+    reverse_domain_name = drs_service_info_id_url_reversal(url=os.environ["HOSTNAME"])
+
+    if len(blueprint.dist) > 0:
+        # Check to see if the information is of type drs. If not, use the available information to return DRS compliant service information
+        for dist in blueprint.dist:
+            if (
+                "type" in dist
+                and isinstance(dist["type"], dict)
+                and "artifact" in dist["type"]
+                and dist["type"]["artifact"] == "drs"
```
Comment (paulineribeyre) [inline indexd/drs/blueprint.py:33]:
```suggestion
                and isinstance(dist.get("type"), dict)
                and dist["type"].get("artifact") == "drs"
```

File: indexd/drs/blueprint.py [hunk 5]
```diff
@@ -1,14 +1,65 @@
...
+            ):
+                drs_dist = dist
```
Comment (paulineribeyre) [inline indexd/drs/blueprint.py:35]: do we need to `break` after this line?

File: indexd/drs/blueprint.py [hunk 6]
```diff
@@ -1,14 +1,65 @@
...
+    ret = {
+        "id": drs_dist.get("id", reverse_domain_name),
+        "name": drs_dist.get("name", "DRS System"),
+        "version": drs_dist.get("version", "1.0.0"),
+        "type": {
+            "group": drs_dist.get("group", "org.ga4gh"),
+            "artifact": drs_dist.get("artifact", "drs"),
+        },
+        "organization": {
+            "name": "Gen3",
```
Comment (paulineribeyre) [inline indexd/drs/blueprint.py:48]: maybe this should be "CTDS"? and this default might not work for all Commons 🤔 can we make it configurable?

File: indexd/drs/blueprint.py [hunk 7]
```diff
@@ -1,14 +1,65 @@
...
+        for dist in blueprint.dist:
+            if (
+                "type" in dist
+                and isinstance(dist["type"], dict)
+                and "artifact" in dist["type"]
+                and dist["type"]["artifact"] == "drs"
+            ):
+                drs_dist = dist
+        if drs_dist == {}:
+            drs_dist = blueprint.dist[0]
```
Comment (paulineribeyre) [inline indexd/drs/blueprint.py:37]: This doesn't seem right, the first "dist" will probably be a different instance of indexd 🤔

File: indexd/default_settings.py [hunk 8]
```diff
@@ -48,7 +49,17 @@
         "name": "DRS System",
         "host": "https://example.com/api/ga4gh/drs/v1/",
         "hints": [],
-        "type": "drs",
+        "type": {
```
Comment (paulineribeyre) [inline indexd/default_settings.py:52]: i'm not sure this is the best place to store this information 🤔 was there a feature doc about this? indexd's `dist` refers to external systems or other indexd instances. Why would it contain information about the current system? I think we need a new config to store this information?

Comment (BinamB) [reply indexd/default_settings.py:52]: Ohhh i didnt realize thats what it was. I thought it was the current distribution information of indexd. I can add it to a different config. There was no feature doc on this. Was kind of a last miniute thing we needed for the ga4gh connect.

**Review by MichaelLukowski — APPROVED** [review]
LGTM

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**